src/Microsoft.OData.Core/ODataMetadataOutputContext.cs

```diff
@@ -267,7 +268,13 @@ protected override async ValueTask DisposeAsyncCore()
         private void WriteMetadataDocumentImplementation()
         {
             IEnumerable<EdmError> errors;
-            if (!CsdlWriter.TryWriteCsdl(this.Model, this.xmlWriter, CsdlTarget.OData, out errors))
+
+            CsdlXmlWriterSettings settings = new CsdlXmlWriterSettings
+            {
+                EdmLibraryCompatibility = (this.MessageWriterSettings.LibraryCompatibility == ODataLibraryCompatibility.UseLegacyVariableCasing) ? EdmLibraryCompatibility.UseLegacyVariableCasing : 0
+            };

```

Comment:
I'm not sure what this assignment is trying to achieve. `MessageWriterSettings.LibraryCompatibility` could have a lot more enum values set in addition to `ODataLibraryCompatibility.UseLegacyVariableCasing`, yet here we're checking whether it's equal to `UseLegacyVariablecasing`.

I think you only need to check whether that particular flag is set:

```suggestion
            CsdlXmlWriterSettings settings = new CsdlXmlWriterSettings
            {
                EdmLibraryCompatibility = (this.MessageWriterSettings.LibraryCompatibility & ODataLibraryCompatibility.UseLegacyVariableCasing == ODataLibraryCompatibility.UseLegacyVariableCasing) ? EdmLibraryCompatibility.UseLegacyVariableCasing : 0
            };
```

Alternatively (more robust in case you have more flags to set, arguably more preferable):

```suggestion
CsdlXmlWriterSettings settings = new CsdlXmlWriterSettings();

if (this.MessageWriterSettings.LibraryCompatibility & ODataLibraryCompatibility.UseLegacyVariableCasing == ODataLibraryCompatibility.UseLegacyVariableCasing)
{
     settings.EdmLibraryCompatibility |= EdmLibraryCompatibility.UseLegacyVariableCasing;
}
```

Reply:
I'm not sure why this doesn't work.